src/util/indexrange.h

```diff
@@ -24,29 +24,29 @@ class IndexRange final: private std::pair<SINT, SINT> {
         DEBUG_ASSERT(empty());
     }
 
-    static IndexRange between(SINT head, SINT tail) {
-        return IndexRange(head, tail);
+    static IndexRange between(SINT start, SINT end) {
+        return IndexRange(start, end);
     }
-    static IndexRange forward(SINT head, SINT length) {
+    static IndexRange forward(SINT start, SINT length) {
         DEBUG_ASSERT(length >= 0);
-        return IndexRange(head, head + length);
+        return IndexRange(start, start + length);
     }
-    static IndexRange backward(SINT head, SINT length) {
+    static IndexRange backward(SINT start, SINT length) {
         DEBUG_ASSERT(length >= 0);
-        return IndexRange(head, head - length);
+        return IndexRange(start, start - length);
     }
 
-    // The near boundary (inclusive)
-    SINT head() const {
+    // The first index within this range (inclusive)
+    SINT start() const {
         return first;
     }
-    // The opposite boundary (exclusive)
-    SINT tail() const {
+    // The next index beyond this range (exclusive)
+    SINT end() const {
         return second;
     }
 
     bool empty() const {
```

Comment:
On the other hand STL uses empty() everywhere and Qt adopts this usage for compatibility in QList for example.